plugins/woocommerce-admin/client/products/fills/more-menu-items/delete-variation-menu-item.tsx

```diff
@@ -0,0 +1,160 @@
+/**
+ * External dependencies
+ */
+import { useParams } from 'react-router-dom';
+import { MenuGroup, MenuItem } from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { useMemo, useState } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import {
+	EXPERIMENTAL_PRODUCT_VARIATIONS_STORE_NAME,
+	ProductVariation,
+	ProductVariationAttribute,
+} from '@woocommerce/data';
+import { getNewPath, navigateTo } from '@woocommerce/navigation';
+import { RemoveConfirmationModal } from '@woocommerce/product-editor';
+import { recordEvent } from '@woocommerce/tracks';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore No types for this exist yet.
+// eslint-disable-next-line @woocommerce/dependency-group
+import { useEntityId, useEntityProp } from '@wordpress/core-data';
+
+export type DeleteVariationMenuItemProps = { onClose(): void };
+
+export const DeleteVariationMenuItem = ( {
+	onClose,
+}: DeleteVariationMenuItemProps ) => {
+	const [ showModal, setShowModal ] = useState( false );
+
+	const { productId } = useParams();
+
+	const variationId = useEntityId( 'postType', 'product_variation' );
```

Comment:
Just thinking out loud:
We are going to need a near similar item at some point for deleting a product. Could we largely re-use this component for this? What areas would have to change to make this more dynamic across multiple product types? Or is variations to much of a unique case?